services/s3/src/it/java/software/amazon/awssdk/services/s3/presignedurl/AsyncPresignedUrlManagerTestSuite.java

```diff
@@ -0,0 +1,287 @@
+        futures.add(presignedUrlManager.getObject(
+            createRequestForKey(concurrentTestKey, "bytes=9-17"),  // "9ABCDEFGH"
+            AsyncResponseTransformer.toBytes()));
+        futures.add(presignedUrlManager.getObject(
+            createRequestForKey(concurrentTestKey, "bytes=18-26"), // "IJKLMNOPQ"
+            AsyncResponseTransformer.toBytes()));
+        futures.add(presignedUrlManager.getObject(
+            createRequestForKey(concurrentTestKey, "bytes=27-35"), // "RSTUVWXYZ"
+            AsyncResponseTransformer.toBytes()));
+
+        CompletableFuture<Void> allFutures = CompletableFuture.allOf(
+            futures.toArray(new CompletableFuture[0]));
+        allFutures.get(30, TimeUnit.SECONDS);
+
+        StringBuilder result = new StringBuilder();
+        for (CompletableFuture<ResponseBytes<GetObjectResponse>> future : futures) {
+            result.append(future.get().asUtf8String());
+        }
+
+        assertThat(result.toString()).isEqualTo("0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ");
+    }
+
+    @Test
+    void getObject_withLargeObjectToFile_savesCompleteContentAndCollectsMetrics() throws Exception {
+        List<MetricCollection> collectedMetrics = new ArrayList<>();
+        MetricPublisher metricPublisher = new MetricPublisher() {
+            @Override
+            public void publish(MetricCollection metricCollection) {
+                collectedMetrics.add(metricCollection);
+            }
+            @Override
+            public void close() {}
+        };
+
+        try (S3AsyncClient clientWithMetrics = s3AsyncClientBuilder()
+                                                           .overrideConfiguration(o -> o.addMetricPublisher(metricPublisher))
+                                                           .build()) {
+            
+            AsyncPresignedUrlManager metricsManager = clientWithMetrics.presignedUrlManager();
+            PresignedUrlGetObjectRequest request = createRequestForKey(testLargeObjectKey);
+            Path downloadFile = temporaryFolder.resolve("large-download-with-metrics-" + UUID.randomUUID() + ".bin");
+            
+            CompletableFuture<GetObjectResponse> future =
+                metricsManager.getObject(request, downloadFile);
+            GetObjectResponse response = future.get(60, TimeUnit.SECONDS);
+            
+            assertThat(response).isNotNull();
+            assertThat(downloadFile).exists();
+            assertThat(downloadFile.toFile().length()).isEqualTo(testLargeObjectContent.length);
+            assertThat(collectedMetrics).isNotEmpty();
+        }
+    }
+
+    static Stream<Arguments> basicFunctionalityTestData() {
+        return Stream.of(
+            Arguments.of("getObject_withValidUrl_returnsContent", 
+                        testGetObjectKey, testObjectContent),
+            Arguments.of("getObject_withValidLargeObjectUrl_returnsContent", 
+                        testLargeObjectKey, null)
```

Comment:
Can we validate checksum for large object? https://github.com/aws/aws-sdk-java-v2/blob/master/utils/src/main/java/software/amazon/awssdk/utils/Md5Utils.java#L70